File: solutions/java/proverb/1/src/main/java/Proverb.java
class Proverb {

    private String proverb;

    Proverb(String[] words) {

        StringBuilder sb = new StringBuilder();
        for (int i = 0, n = words.length - 1; i < n; ++i) {
            sb.append("For want of a " + words[i] + " the " + words[i + 1] + " was lost.\n");
        }
        if (words.length > 0) {
            sb.append("And all for the want of a " + words[0] + ".");
        }
        this.proverb = sb.toString();
    }

    String recite() {
        return this.proverb;
    }

}

File: solutions/python/proverb/1/proverb.py
def proverb(*input_iterable, qualifier=None):
    len_iterable = len(input_iterable)
    if len_iterable == 0:
        return []
    out = []
    for i in range(1, len_iterable):
        text = f"For want of a {input_iterable[i-1]} the {input_iterable[i]} was lost."
        out.append(text)
    qualifier_text = ("" if qualifier is None else f"{qualifier} ") + input_iterable[0]
    out.append(f"And all for the want of a {qualifier_text}.")
    return out